### DISCUSSION TITLE
What to do now that Heroku free tiers are gone?

### DISCUSSION THREAD

**Discussion by cirosantilli** [realworld-apps/realworld#974] [DEMO / API health status]
Yo @geromegrignon what does the reference backend run on BTW, and is it free/who pays?

I wonder if a project with this many stars would be able to get some free tier accounts from Heroku. 

With Heroku shutting down free dynos in October https://blog.heroku.com/next-chapter , I think my implementation is going to go down for example: https://cirosantilli-realworld-next.herokuapp.com/ I don't think I'll be paying 7 dollars month to keep it alive, and not in the mood to port to another service that might shut down/kill free tiers later on anyways...

It would be good for the project as a hole if we could have such a charitable willing to give some CPU cycles in exchange for some well deserved publicity.

https://blog.heroku.com/next-chapter says

> [Students and Nonprofit Program](https://blog.heroku.com/next-chapter#students-and-nonprofit-program)
We appreciate Heroku’s legacy as a learning platform. Many students have their first experience with deploying an application into the wild on Heroku. Salesforce is committed to providing students with the resources and experiences they need to realize their potential. We will be announcing more on our student program at [Dreamforce](https://www.salesforce.com/dreamforce/). For our nonprofit community, we are working closely with our nonprofit team, too.

https://help.heroku.com/VF2KIQ5S/does-heroku-offer-non-profit-or-educational-discounts says potentially outdated information:

> Presently, Heroku does not have any discount offerings, even for non-profit or educational projects. However we do already offer extensive discounts with [Free Dynos](https://www.heroku.com/pricing), and the many free add-ons available on our [Add-on Marketplace](https://elements.heroku.com/addons).
>
> You may also be interested to know that our parent company does have a formal discount program for non-profits through the [Salesforce Foundation](http://www.salesforcefoundation.org/power-of-us/).

**Comment by geromegrignon** [comment]
Hi @cirosantilli,
The backend runs freely on Heroku (but the database is hosted on Supabase for monitoring purposes).
As we only use one app on Heroku, 1000 free dyno hours (with a verified account) is enough to keep it free (as long there is a free plan).
Besides some bugs and the hack they faced (leading to GitHub integration interruption), it went well so far.

I currently investigate Heroku alternatives to keep it free.
That's not so easy for the reference backend as it runs all day long and most alternatives have morel imitations (making it to keep a busy server online).
The non-Profit program is for legal organizations only (with a tax id number, for example) so RealWorld doesn't meet the requirements.

As there are free alternatives, I'd instead encourage direct GitHub sponsoring for implementation maintainers rather than having to pay for servers. Given the current number of existing implementations, it would be be a mess to triage worthy implementations and require additional maintenance effort in creating and managing rules.

**Comment by geromegrignon** [comment]
It doesn't mean I'm not open to solutions though.
DigitalOcean might provide a solution: https://www.digitalocean.com/open-source/credits-for-projects

To sump up the requirements:

What the reference backend currently needs:
- to be live 24/7 (around 700 hours monthly). free with a verified Heroku account (limited to 500 with Railway)

What the reference backend will additionally need in the future (due to upload feature request):
- data storage. Limited to 1GB with current Supabase free plan used.

> Whatever solution is used, it'll more likely rely on regular reset (considering there are 100 778 users created so far since last November).

About being sponsored (globally, not related to GitHub sponsorship):

The usage:
- as we value each new implementation, the sponsorship won't allow hosting all backend ones.
- database and object storage would be shared across the hosted backend to reduce costs.

The project:
- I'll speed up the new version release as a more modern 'real world' project is a requirement in my opinion to make this project valuable in late 2022 for a sponsor.
- sponsored implementations would require higher expectations, regular maintenance, and maintenance right access for the RealWorld maintenance team.


That's mostly about backends but could apply with frontends too.
